boards/arm/mps2_an521/mps2_an521_nonsecure.dts

```diff
@@ -77,8 +77,8 @@
 		reg = <0x28100000 0x100000>;
 	};
 
-	flash0: flash@100400 {
-		reg = <0x100400 0xDF00000>;
+	flash0: flash@100000 {
+		reg = <0x100000 0xDF00000>;
```

Comment:
Together with this change you need to modify the way the non-secure image is signed because now `zephyr.bin` will contain the gap resulting from `ROM_START_OFFSET` at its beginning.
You need to add `--included-header` in this invocation of `imgtool.py`: https://github.com/zephyrproject-rtos/zephyr/blob/2cd556a689e36c5cf19336dcacafb41b900175e6/boards/arm/mps2_an521/CMakeLists.txt#L52-L63